pkg/machinery/resources/secrets/os_root.go

```diff
@@ -33,6 +33,7 @@ type OSRootSpec struct {
 	AcceptedCAs     []*x509.PEMEncodedCertificate     `yaml:"acceptedCAs" protobuf:"5"`
 	CertSANIPs      []netip.Addr                      `yaml:"certSANIPs" protobuf:"2"`
 	CertSANDNSNames []string                          `yaml:"certSANDNSNames" protobuf:"3"`
+	Rsa             bool                              `yaml:"rsa" protobuf:"6"`
```

Comment:
I don't think we should have this field ever, as it's already in `IssuingCA` indirectly

Reply:
Worker nodes don't have `IssuingCA` at all - otherwise they would have issued a certificate for themselves.

Reply:
The configs that talosctl generates give the workers an `IssuingCA` but with only the `Crt` set, and `Key` empty. 
The worker then appends that `Crt` to `AcceptedCAs` and clears `IssuingCA`: https://github.com/siderolabs/talos/blob/c55af59a2583b12d4953c8459fc52f7b22cae9c5/internal/app/machined/pkg/controllers/secrets/root.go#L164-L173

My change stores the signature algorithm of said `Crt` in the `OSRootSpec` struct so that this information can later be used to create the appropriate CSR: https://github.com/gwenya/talos/blob/a3672d11482c29e6d798f397d1a329cde6ed6ed3/internal/app/machined/pkg/controllers/secrets/root.go#L170-L175

Reply:
Storing algorithm sounds better now, but we should probably use proper type to make it look nice in the resource itself, I can work it out later on.

Reply:
I initially used the `x509.SignatureAlgorithm` type from the standard library, but `make docs` failed to generate protobuf for that, so I switched it to `int`. If there is some way to make that work I would prefer it.